SpatialGDK/Documentation/content/non-unreal-layers.md

```diff
@@ -0,0 +1,213 @@
+```
+
+which are accessible via a reference to the net driver:
+
+```
+SpatialWorkerConnection* Connection = Cast<USpatialNetDriver>(World->GetNetDriver())->Connection;
+```
+
+There is a basic example in the section below, and for more examples of how to construct component updates, command requests, and more, check the [C API page on serialization](https://docs.improbable.io/reference/latest/capi/serialization).
+
+### Receiving data
+
+You can receive components updates through implementing dispatcher callbacks for specified component IDs and op types. These callbacks are detected by the Unreal dispatcher pipeline on initialization and used instead of the existing pipeline where the component ID and op type are matched.
+
+To do this, you need to create a class derived from the `UOpCallbackTemplate` base class and implement the `GetComponentId` and one or more of the callback methods from the. These callback methods are parameterized by C API op types:
+
+```
+virtual void OnAddComponent(const Worker_AddComponentOp& op) {}
+virtual void OnRemoveComponent(const Worker_RemoveComponentOp& op) {}
+virtual void OnComponentUpdate(const Worker_ComponentUpdateOp& op) {}
+virtual void OnAuthorityChange(const Worker_AuthorityChangeOp& op) {}
+virtual void OnCommandRequest(const Worker_CommandRequestOp& op) {}
+virtual void OnCommandResponse(const Worker_CommandResponseOp& op) {}
+  ```
+
+There is a basic example in the section below, and for more examples of how to deserialize C API types, check the [C API page on serialization](https://docs.improbable.io/reference/latest/capi/serialization).
+
+### Adding to snapshots
+
+You can customize snapshot generation through creating a class derived from the `USnapshotGenerationTemplate` base class, and implementing the following method:
+
+```
+/**
+	* Write to the snapshot generation output stream
+	* @param OutputStream the output stream for the snapshot being created.
+	* @param NextEntityId the next available entity ID in the snapshot, this reference should be incremented appropriately.
+	* @return bool the success of writing to the snapshot output stream, this is returned to the overall snapshot generation.
+	**/
+bool WriteToSnapshotOutput(Worker_SnapshotOutputStream* OutputStream, Worker_EntityId& NextEntityId);
+```
+
+## Example
+
+Below is a simple, example schema file which could be used by a separate worker layer to track player statistics:
+
+```
+package improbable.session;
+
+type MyRequest {
+  string player_name = 1;
+}
+type MyResponse {}
+
+component Session {
+    id = 1337;
+    uint32 player_count = 1;
+    command MyResponse some_command(My_Request);
+}
+```
+
+#### Sending data
+
+You could serialize and send a component update in user code in the following way:
+
+```
+void SendSomeUpdate(Worker_EntityId target_entity_id, Worker_ComponentId component_id) {
+    Worker_ComponentUpdate component_update;
```

Comment:
Nit: we should probably consistently use the Unreal naming convention (`UpperCamelCase`, prefixed with `b` for bools) for variables in these snippets (since most Unreal user code will use the same)

Reply:
ty yeah, docs review in process with a doc writer in separate google doc, have updated snippet 👍